src/qmapshack/poi/CPoiPOI.cpp

```diff
@@ -126,6 +127,43 @@ void CPoiPOI::draw(IDrawContext::buffer_t& buf)
     mutex.unlock();
 }
 
+void CPoiPOI::findPOICloseBy(const QPoint &px, poi_t & poiItem) const
+{
+    QPointF p = px;
+    poi->convertPx2Rad(p);
+    for(const QString& category : categoryActivated.keys(Qt::Checked))
+    {
+        int minLonM10 = qFloor(p.x() * RAD_TO_DEG * 10);
+        int minLatM10 = qFloor(p.y() * RAD_TO_DEG * 10);
+        //The tile that contains the POI has to be loaded, since a position on screen is given
+        for(const rawPoi_t& poiItemFound : loadedPOIs[category][minLonM10][minLatM10])
+        {
+            QPointF pt = poiItemFound.coordinates;
+            poi->convertRad2Px(pt);
+            QPointF x = px - pt;
+            if(x.manhattanLength() < 10)
+            {
+                poiItem.pos = poiItemFound.coordinates;
+                for(auto tag : {"name=", "brand=", "operator="})
+                {
+                    auto matches = poiItemFound.data.filter(tag);
```

Comment:
again `auto` is a bad choice. Everyone has to lookup the type of data and the return type of filter to understand that auto is in fact a `QStringList`. It should be a `const QStringList&` and I am not sure if auto does that.